core/scripts/vrfv2plus/testnet/README.md

```diff
@@ -0,0 +1,246 @@
+# Using the External Subscription Owner Example
+
+The [external subscription owner example contract](../../../../contracts/src/v0.8/tests/VRFExternalSubOwnerExample.sol)
+allows its owner to request random words from VRF V2 if it is added as a
+consumer for a funded VRF subscription.
+
+This guide covers:
+ 1. Deploying the contract
+ 2. Creating, funding, checking balance, and adding a consumer to a VRF V2 
+    subscription
+ 3. Requesting randomness from the contract
+
+## Setup
+
+Before starting, you will need:
+1. An EVM chain endpoint URL
+2. The chain ID corresponding to your chain
+3. The private key of an account funded with LINK, and the chain's native token
+   (to pay transaction fees)
+4. [The LINK address, VRF coordinator address, and key hash](https://docs.chain.link/docs/vrf/v2/supported-networks/)
+   for your chain.
+5. [Go](https://go.dev/doc/install)
+
+The endpoint URL can be a locally running node, or an externally hosted one like
+[alchemy](https://www.alchemy.com/). Your chain ID will be a number
+corresponding to the chain you pick. For example the Rinkeby testnet has chain
+ID 4. Your private key can be exported from [MetaMask](https://metamask.zendesk.com/hc/en-us/articles/360015289632-How-to-Export-an-Account-Private-Key).
+
+Note: Be careful with your key. When using testnets, it's best to use a separate
+account that does not hold real funds.
+
+Run the following command to set up your environment:
+
+```shell
+export ETH_URL=<YOUR ETH URL>
+export ETH_CHAIN_ID=<YOUR CHAIN ID>
+export ACCOUNT_KEY=<YOUR PRIVATE KEY>
+export LINK=<LINK ADDRESS>
+export LINK_ETH_FEED=<ADDRESS OFF LINK/ETH FEED>
+export COORDINATOR=<COORDINATOR ADDRESS>
+export KEY_HASH=<KEY HASH>
+export ORACLE_ADDRESS=<YOUR ORACLE NODE ADDRESS>
+export PUB_KEY=<YOUR UNCOMPRESSED PUBLIC KEY>
+```
+
+By default, the script automatically estimates gas limits for operations. Optionally, `ETH_GAS_LIMIT_DEFAULT` environment variable can be set to override gas limit for operations. 
+
+Now "cd" into the VRF V2 testnet scripts directory:
+
+```shell
+cd <YOUR LOCAL CHAINLINK REPO>/core/scripts/vrfv2/testnet
+```
+
+## Deploying a full VRF Universe (BHS, Registered + Funded Coordinator, Consumer)
+
+To deploy a full VRF environment on-chain, run:
+
+```shell
+go run . deploy-universe --link-address=$LINK --link-eth-feed=$LINK_ETH_FEED --subscription-balance=<SUBSCRIPTION AMOUNT> --uncompressed-pub-key=$PUB_KEY --oracle-address=$ORACLE_ADDRESS
```

Comment:
Maybe worth stating here that not specifying `link-address` and `link-eth-feed` will make the script deploy fresh versions of these two things, which is fine for testing most things.

Reply:
Added this message